jdaviz/core/helpers.py

```diff
@@ -314,32 +315,43 @@ def show(self, loc="inline", title=None):
         If "sidecar" is requested in the "classic" Jupyter notebook, the app will appear inline,
         as only JupyterLab has a mechanism to have multiple tabs.
         """
-        try:
-            if loc == "inline":
+        # Check if the user is running Jdaviz in the correct environments.
+        # If not, provide a friendly msg to guide them!
+        if get_ipython().__class__.__name__ != 'ZMQInteractiveShell':
+            boilerplate_e = \
+                RuntimeError("\nYou are currently running Jdaviz from an unsupported "
+                             "shell. Jdaviz is intended to be run within a Jupyter "
+                             "notebook, or directly from the command line.\n\n"
+                             "To run from Jupyter, call <your viz>.show() from a notebook cell.\n"
+                             "To see how to run from the command line, run: "
+                             "'jdaviz --help' outside of Python.\n\n"
+                             "To learn more, see our documentation at: "
+                             "https://jdaviz.readthedocs.io\n\n"
+                             "Thanks for trying out Jdaviz! :)")
+            raise boilerplate_e
+
+        if loc == "inline":
```

Comment:
Aren't these already in `main`? Why I still see this as diff here? Does this PR need a rebase or did I miss something?

https://github.com/spacetelescope/jdaviz/blob/2386e460b35a1ad77c5490e582a143153ab64b03/jdaviz/core/helpers.py#L318

Reply:
I think just because of the indentation change (and removing the outer try/except)? 🤷

Reply:
A rebase would be nice if this hasn't been rebased in a while, just to be on the safe side.

Reply:
On it!

Reply:
Rebased and this remains. I believe Kyle's right; this is due to the removal of the upper try block